run-tests.py

```diff
@@ -86,6 +86,15 @@ def run_doc_tests():
             ],
             check=True,
         )
+        print("Checking for doctests.")
```

Comment:
Could you remove the subprocess above this and [configure `pytest-doctestplus` to run for docs-related files](https://github.com/scientific-python/pytest-doctestplus#setup-and-configuration) as well? Furthermore, if we get that done, we don't need the Sphinx version of it, so you can then remove `sphinx.ext.doctest` and `doctest_global_setup` from `docs/conf.py`.

Reply:
Isn't the above subprocess for building documentation? How to test documentation builds if we remove this?

Reply:
> Isn't the above subprocess for building documentation?

The documentation is built using Sphinx's HTML builder (`-b html`) while this is the `doctest` builder (`-b doctest`). The doctest builder is enabled by the `doctest` extension, see https://www.sphinx-doc.org/en/master/usage/builders/index.html#builders

> How to test documentation builds if we remove this?

Good question. We can do that with `nox -s docs` instead, which runs separate commands in CI versus locally – we never added the HTML builder to `run-tests.py` I think and we always kept it in `noxfile.py` (and earlier in `tox.ini`). We will either completely remove `run-tests.py` or clear out major parts of it anyway, so reducing repetition will be nice in the coming weeks.

Reply:
This is really helpful! Thanks. However `--doctest-plus` cannot recognise tests in `docs`. Could it be because of [this](https://github.com/scientific-python/pytest-doctestplus#setup-and-configuration:~:text=Sphinx%20Doctest%20Directives) directive?

Reply:
I don't think we have many (or any, AFAIR) doctests inside `docs/`. We have most of them in the docstrings for the public API classes and methods, and they are in the main `pybamm/` folder – as long as they are being tested and they pass, we should be good.

Reply:
Running pytest on docs Fails with exit code 5. Pytest treats `no test run` as [error](https://pytest.org/en/stable/reference/exit-codes.html#:~:text=Exit%20code%205,tests%20were%20collected). 
This stops running further tests.
We can maybe remove running tests on docs for now until we have any tests written. Is there a way to suppress this error? One hackish way could be to add a dummy test somewhere in the docs folder.

Reply:
Why do we need to run `pytest` on the `docs/` folder, though?

Reply:
Aren't the docs-related files inside the `docs` folder? Am I missing something here?

Reply:
Sorry, I somehow missed the context of the comment I posted originally. All the doctests are in `pybamm/`, inside the docstrings for the public API (marked with `.. doctest::`), so we don't need to run doctests across `docs/` – we don't have anything to test there. 

Therefore, from my earlier comment:

> and [configure pytest-doctestplus to run for docs-related files](https://github.com/scientific-python/pytest-doctestplus#setup-and-configuration) as well?

this isn't needed. We just need to remove `sphinx.ext.doctest` and `doctest_global_setup`, and the Sphinx subprocess above, and this comment can be resolved.

Reply:
This clears the confusion. Thanks!